### PR TITLE
build: shadow jar - all deps in one jar

### LINKED ISSUE #226: FatJar in Maven repo
### Prerequisites
* [x] If this is a question/suggestion, I have already considered discuss it on [Discord Server](https://discord.gg/FQ5vgW3)
* [x] This issue specifically has something to do with Twitch4J
* [x] I have attempted to [look for similar issues](https://github.com/twitch4j/twitch4j/issues) already

### Description
Lots of coders (mainly Minecraft coders) asking for `uberJar` thing. The fact can be real if we provide them into next release with #220

### Solutions
Adding `com.github.johnrengelman.shadow` plugin to build for each dependecies. For better performance following by [documentation](https://imperceptiblethoughts.com/shadow/getting-started/) we need build exclude filter for avoid file duplications. I don't know how Kotlin classes will handle it. 

### Alternatives
Didn't even met some other alternatives. This solution is way simple.

### PR DESCRIPTION
**Pull request by stachu540**
### Prerequisites for Code Changes
* [x] This pull request follows the code style of the project
* [x] I have tested this feature

### Changes Proposed
1. Apply plugin `com.github.johnrengelman.shadow` - Closes #226 

### Additional Information 
In #226 where providing this plugin and applying `shaded` word into `archiveClassifier` for `shadowJar` task, will automatically adds into `MavenPublication`. 

All those changes are be tested before the pull request.

### CODE REVIEW

**Review by PhilippHeuer — CHANGES_REQUESTED** [review]
Should we relocate all packages to avoid conflicts with other dependencies and also with ourselves? (ie. to allow ppl to use the shaded TwitchChat 1.0 together with shaded TwitchPubSub 1.1 as single modules)

> https://imperceptiblethoughts.com/shadow/configuration/relocation/#automatically-relocating-dependencies

We would just need to use a unique prefix for each module.

File: deployment.gradle [hunk 1]
```diff
@@ -19,6 +19,14 @@ extensions.configure(JavaPluginExtension) {
...
 }
 
+// Shadow Jar
+shadowJar {
+	archiveClassifier.set('shaded')
+	manifest {
+		inheritFrom jar.manifest
+	}
+}
+
```
Comment (stachu540) [inline deployment.gradle:29]:
```suggestion
task shadowJarRelocate(type: com.github.jengelman.gradle.plugins.shadow.tasks.ConfigureShadowRelocation) {
	target = shadowJar
	prefix - "t4j" // up to you
}

// Shadow Jar
shadowJar {
	dependsOn shadowJarRelocate
	archiveClassifier.set('shaded')
	manifest {
		inheritFrom jar.manifest
	}
}
```

**Review by PhilippHeuer — APPROVED** [review]
Had to change it a little because the example in the shadow documentation wasn't working, but it works now.

**Comment by stachu540** [comment]
#226 Closed

**Pull request merged**